plugins/salesforce-as-datasource-as-eventsource/README.md

```diff
@@ -1 +1,146 @@
+│      │ Name                               │ Description                                                        │
+├──────┼────────────────────────────────────┼────────────────────────────────────────────────────────────────────┤
+│  ◯   │ prisma-as-datastore                │ Prisma as a datasource plugin for Godspeed Framework.              │
+├──────┼────────────────────────────────────┼────────────────────────────────────────────────────────────────────┤
+│  ◯   │ aws-as-datasource                  │ aws as datasource plugin for Godspeed Framework                    │
+├──────┼────────────────────────────────────┼────────────────────────────────────────────────────────────────────┤
+│  ◯   │ excel-as-datasource                │ excel as datasource plugin for Godspeed Framework                  │
+├──────┼────────────────────────────────────┼────────────────────────────────────────────────────────────────────┤
+│  ◯   │ mailer-as-datasource               │ mailer as datasource plugin for Godspeed Framework                 │
+├──────┼────────────────────────────────────┼────────────────────────────────────────────────────────────────────┤
+│ ❯◯   │ salesforce-as-datasource-as-eventsource │ salesforce as datasource-as-eventsource plugin for Godspeed Framework   │
+└──────┴────────────────────────────────────┴────────────────────────────────────────────────────────────────────┘
+```
+
+### Example usage Datasource (Producer):
+
+1. Update configuration file based on your requirements in `Datasource/salesforce.yaml`.
+#### salesforce config ( src/datasources/salesforce.yaml )
+```yaml
+type: salesforce
+loginUrl: 'https://dummy.salesforce.com'
+username: dummy.username.password@organizationname.com
+password: password@123
+```
+
+
+
+#### salesforce event for Producer ( src/events/salesforce_pub.yaml )
+In the event, we establish an HTTP endpoint that accepts parameters such as message content. When this endpoint is invoked, it triggers the `salesforce-publish` function. This function, in turn, takes the provided  message as input arguments and performs the task of publishing the message to the specified salesforce api end-point.
+```yaml
+# event for Publish
+
+'http.post./salesforce-pub':
+  fn: salesforce-publish
+  body:
+    content:
+      application/json:
+        schema:
+          type: object
+          properties:
+            message:
+              type: string
+          required: ['message']
+  responses:
+    200:
+      content:
+        application/json:
+          schema:
+            type: object
+            properties:
+              name:
+                type: string
+
+```
+#### salesforce workflow for Producer ( src/functions/salesforce-publish.yaml )
+
+
+```yaml
+id: salesforce-publish
+summary: salesforce publish message
+tasks:
+  - id: push_data_into_salesforce
+    description: Push data into salesforce
+    fn: datasource.salesforce_api.post./data/v56.0/sobjects/dummy__e
```

Comment:
were we not using the salesforce sdk to publish event? This is an axios salesforce datasource example. If developer can do both ways then mention there are two ways to call salesforce API. 
A. call via axios: in that case you do not need this plugin. For that install axios plugin (insert link to its /docs page) and configure salesforce API as an axios dataasource.
B. Call via this plugin: Example of a sample call to an API via the salesforce SDK is below